src/framework/input/element-input.js

```diff
@@ -736,15 +736,23 @@ class ElementInput {
         if (eventType === 'mouseup' && this._pressedElement) {
             // click event
             if (this._pressedElement === this._hoveredElement) {
-                this._pressedElement = null;
-
-                // fire click event if it hasn't been fired already by the touchup handler
-                if (!this._clickedEntities || !this._clickedEntities[this._hoveredElement.entity.getGuid()]) {
+                // fire click event if it hasn't been fired already by the touchend handler
+                const guid = this._hoveredElement.entity.getGuid();
+                // Always fire, if there are no clicked entities
+                let fireClick = !this._clickedEntities;
+                // But if there are, we need to check how long ago touchend added a "click brake"
+                if (this._clickedEntities) {
```

Comment:
```suggestion
                if (this._clickedEntities.length > 0) {
```
Same as above

Reply:
Same as above, btw it's also an `{}` and not an `[]`